### ISSUE TITLE
libcudnn (R5) not found in library path

### ISSUE THREAD

**Issue by tastyminerals** [soumith/cudnn.torch#254]
Sign, looks like cuda issues are not letting me go.
I have cuda7.5 installed.
```
ldconfig -p | grep cuda
	libicudata.so.55 (libc6,x86-64) => /usr/lib/x86_64-linux-gnu/libicudata.so.55
	libicudata.so.55 (libc6) => /usr/lib/i386-linux-gnu/libicudata.so.55
	libicudata.so (libc6,x86-64) => /usr/lib/x86_64-linux-gnu/libicudata.so
	libcudart.so.7.5 (libc6,x86-64) => /usr/lib/x86_64-linux-gnu/libcudart.so.7.5
	libcudart.so (libc6,x86-64) => /usr/lib/x86_64-linux-gnu/libcudart.so
	libcuda.so.1 (libc6,x86-64) => /usr/lib/x86_64-linux-gnu/libcuda.so.1
	libcuda.so.1 (libc6) => /usr/lib/i386-linux-gnu/libcuda.so.1
	libcuda.so (libc6,x86-64) => /usr/lib/x86_64-linux-gnu/libcuda.so
	libcuda.so (libc6) => /usr/lib/i386-linux-gnu/libcuda.so
```
`LD_LIBRARY_PATH` looks ok to me
```
echo $LD_LIBRARY_PATH 
/usr/lib/x86_64-linux-gnu/:/usr/lib/:/lib
```
However, when I attempt to run lstm network training, I get the following.

```
./torch/install/bin/th /local/pavels/lstm-char-cnn/main.lua -data_dir data/ru -savefile -ru-char-large -use_chars 1 -cudnn 1 -gpuid 1 -EOS '' 
using CUDA on GPU 1...	
using cudnn...	
/local/pavels/torch/install/bin/luajit: /local/pavels/torch/install/share/lua/5.1/trepl/init.lua:384: /local/pavels/torch/install/share/lua/5.1/trepl/init.lua:384: /local/pavels/torch/install/share/lua/5.1/cudnn/ffi.lua:1599: 'libcudnn (R5) not found in library path.
Please install CuDNN from https://developer.nvidia.com/cuDNN
Then make sure files named as libcudnn.so.5 or libcudnn.5.dylib are placed in your library load path (for example /usr/local/lib , or manually add a path to LD_LIBRARY_PATH)

stack traceback:
	[C]: in function 'error'
	/local/pavels/torch/install/share/lua/5.1/trepl/init.lua:384: in function 'require'
	/local/pavels/lstm-char-cnn/main.lua:98: in main chunk
	[C]: in function 'dofile'
	...vels/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:145: in main chunk
	[C]: at 0x00405d70
```
Indeed my system does not have `libcudnn.so.5`, but what package installs this file? I thought `luarocks install cudnn` should be enough.

**Comment by SeanNaren** [comment]
Hey you need to create an account [here](https://developer.nvidia.com/cudnn) (its free!) and download the main library to use these bindings. Follow the instructions given depending on the version you download (recommend downloading cuDNN 5.1 for CUDA 7.5).

The instructions should include something like this after you extract the library file:
```
sudo cp cuda/include/cudnn.h /usr/local/cuda/include
sudo cp cuda/lib64/libcudnn* /usr/local/cuda/lib64
```

**Comment by tastyminerals** [comment]
I don't have a **sudo** access unfortunately.

**Comment by soumith** [comment]
If you dont have sudo access,  just add the path to libcudnn.so to your $LD_LIBRARY_PATH

**Comment by tastyminerals** [comment]
Uncompressed and copied the file to cuda5 dir.
```
[pavels@jones-1 pavels]$ ls /local/pavels/cuda5/lib/
libcudnn.so@  libcudnn.so.5@  libcudnn.so.5.0.5*  libcudnn_static.a
```
Added the path to `$LD_LIBRARY_PATH `, sourced `~/.bashrc` and still getting the same error :(
```
[pavels@jones-1 pavels]$ echo $LD_LIBRARY_PATH 
/usr/lib/x86_64-linux-gnu/:/usr/lib/:/local/pavels/cuda5/lib/:/lib
```

**Comment by tastyminerals** [comment]
Ok, maybe something is wrong with my config. I dropped `-cudnn 1` parameter and it started working without it.

**Comment by shailzajolly** [comment]
hello everyone,

I have tried almost all the comments said here but still getting same error.

/home/ubuntu/torch/install/bin/luajit: /home/ubuntu/torch/install/share/lua/5.1/trepl/init.lua:389: /home/ubuntu/torch/install/share/lua/5.1/trepl/init.lua:389: /home/ubuntu/torch/install/share/lua/5.1/cudnn/ffi.lua:1599: 'libcudnn (R5) not found in library path.
Please install CuDNN from https://developer.nvidia.com/cuDNN
Then make sure files named as libcudnn.so.5 or libcudnn.5.dylib are placed in your library load path (for example /usr/local/lib , or manually add a path to LD_LIBRARY_PATH)

stack traceback:
	[C]: in function 'error'
	/home/ubuntu/torch/install/share/lua/5.1/trepl/init.lua:389: in function 'require'
	eval.lua:48: in main chunk
	[C]: in function 'dofile'
	...untu/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:150: in main chunk
	[C]: at 0x00405d50

**Comment by krt-ghelia** [comment]
hello, every one.

In @SeanNaren 's comment, we can get libcudnn.so.5.
I installed cuDNN7 & CUDA9.0 before, so libcudnn.so.7(?) should be in /usr/local/cuda/lib64. But I can't find.
Before, I have trained CNNs with chainer, 
at that time, I tried these in iPython:
`import chainer`
`print(chainer.cuda.available)` 
returns True, (this should mean I can use CUDA.)
`print(chained.cuda.cudnn_enabled`
returns True. (this should mean I have installed cuDNN and it's set enable.)

These seems to mean I can use cuDNN, but error message told me that there is no libcudnn.so.X.
what makes it wrong??
(I'm studying English, so sentences above may be somehow wrong. Sorry.)
